nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js

```diff
@@ -1,14 +1,59 @@
 const express = require("express");
 const router = express.Router();
 
-const meals = require("./../data/meals.json");
+const meals = require("../data/meals.json");
 
 router.get("/", async (request, response) => {
-  try {
-    console.log(meals);
-    console.log("in /api/meals");
-  } catch (error) {
-    throw error;
+  console.log(meals);
+  console.log("in /api/meals");
+  const query1 = request.query;
+  console.log(request.query);
+
+  if (
+    (query1.maxPrice && isNaN(parseInt(query1.maxPrice))) ||
+    (query1.createdAfter && isNaN(Date.parse(query1.createdAfter))) ||
+    (query1.limit && isNaN(parseInt(query1.limit)))
+  ) {
+    return response.status(400).send("Given parameter is not supported");
+  }
+
+  if (query1.maxPrice || query1.title || query1.createdAfter || query1.limit) {
+    let updatedMeals = meals
+      .filter((meal) =>
+        query1.maxPrice ? meal.price <= query1.maxPrice : true
+      )
+      .filter((meal) =>
+        query1.title ? meal.title.toLowerCase().includes(query1.title) : true
```

Comment:
Good that you used toLowerCase() to compare the two strings, it's a good practice to do that.
However, don't forget to do it for query1.title as well, otherwise it won't work 100% correctly!